cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsClient.java

```diff
@@ -50,6 +48,9 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AwsClient.class);
 
+    // Default retries is 3. This allows for more time for backoff during throttling
+    private static final int MAX_CLIENT_RETRIES = 30;
```

Comment:
is it retries every operation where there is an error code in the response, im wondering is this valid then (like with the retry clients we are catching specifically the rate limit issues)

Reply:
This uses https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/retry/PredefinedRetryPolicies.html which uses the default backoff and retry conditions.

https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/retry/PredefinedRetryPolicies.html#DEFAULT_RETRY_CONDITION

This highlights that for throttling and other retriable errors are retried. Its not for any error, but for ones that are AWS service side and we want to retry for. The ones that we may hit more often would be the throttling unless AWS has some big issue.